bluemira/equilibria/coils/_grouping.py

```diff
@@ -998,3 +996,25 @@ def _sum(
         ind = self._control_ind if control else slice(None)
 
         return np.sum(output[..., ind], axis=-1) if sum_coils else output[..., ind]
+
+    def get_coiltype(self, ctype):
+        """Get coils by coils type"""
+        if coiltype := self._get_coiltype(ctype):
+            return CoilSet(*coiltype)
+        return None
+
+    @classmethod
+    def from_group_vecs(cls, eqdsk: EQDSKInterface):
+        """Same as its parent class,
+        initialize an instance of CoilSet from group vectors,
+        but it also auto-populate the control attribute.
+
+        This is done so that the user doesn't have to
+        manually set passive coils as non-controlled.
+        """
+        self = super().from_group_vecs(eqdsk)
+
+        if coil_name := self.get_coiltype(CoilType.NONE):
+            # if passive coils are present: give them the correct names.
+            self.control = coil_name.name
```

Comment:
We need another line for dummy coils
```suggestion
        if (none_coil := self.get_coiltype(CoilType.NONE)) or (
            dum_coil := self.get_coiltype(CoilType.DUM)
        ):
            # if passive coils are present: give them the correct names.
            self.control = [
                *([] if none_coil is None else none_coil.name),
                *([] if dum_coil is None else dum_coil.name),
            ]
            import
```

Reply:
Should be the negation of this. Will implement accordingly.